tests/mqtt/test_adapter.py

```
@@ -0,0 +1,769 @@
+
+class Test_Logging_Connection_Result(unittest.TestCase):
+
+    def setUp(self) -> None:
+        self.adapter = MQTTClientAdapter(
+            "some_company", "test_car", 0.5, TEST_ADDRESS, TEST_PORT, EventQueue()
+        )
+
+    def test_error_is_logged_when_connecting_to_broker_fails(self):
+        # broker does not exist
+        with self.assertLogs(logger=_logger._logger, level=logging.ERROR):
+            with self.assertRaises(ConnectionRefusedError):
```

Ruff

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)